### PR TITLE
Only include peer in prefix for single hosts

### PR DESCRIPTION
**Pull request by TheNaterz**
Currently, when a status message gets printed for a module that has an RHOSTS defined as a file with hundreds of hosts, the prefix takes this gigantic space separated list of hosts and shoves it into a message. Every time the message is printed, you get this really ugly block of text and it makes the output hard to read.

I propose that peer information isn't necessary enough to include in these messages.

## Before

- [ ] Start `msfconsole`
- [ ] `use auxiliary/scanner/ssh/ssh_version` (just as an example)
- [ ] Set RHOSTS to some file with 100 hosts or something.
- [ ] Every scanner progress message looks gross.

## After
- [ ] Do all the above steps.
- [ ] Bask in the glory of minimal output. Look upon its simplistic face and behold its overwhelming mediocrity. But also it's cleaner and cleanliness is close to godliness.

In other words...

When did this:
selection_002
become hotter than this?:
selection_001

Thoughts?

### CODE REVIEW

**Comment by wvu-r7** [comment]
My thoughts are death to the [peer gods](https://github.com/rapid7/metasploit-framework/pull/6526).

**Comment by busterb** [comment]
🙏 📿

**Comment by jhart-r7** [comment]
How does this affect `RHOSTS` use-cases that are not `file://` based?

**Comment by busterb** [comment]
Any other mixins we should update? udp?

File: lib/msf/core/exploit/tcp.rb [hunk 1]
```diff
@@ -196,11 +196,12 @@ def cleanup
   end
 
   def print_prefix
```
Comment (busterb) [inline lib/msf/core/exploit/tcp.rb]: I think deleting print_prefix entirely from this file has the same outcome as it just calling super.

**Comment by TheNaterz** [comment]
Maybe? This has just been the real nagging one that I've noticed. If this issue propagates in other areas, I would recommend the same changes, yeah.

File: lib/msf/core/exploit/tcp.rb (hunk 1 above)
Comment (TheNaterz) [reply lib/msf/core/exploit/tcp.rb]: Yep valid!

**Comment by wvu-r7** [comment]
It propagates so hard we considered reverting the entire changeset!

**Comment by jhart-r7** [comment]
IMO, peer information is VERY useful, if not critical, to any `RHOSTS` use-case involving more than one host.  How are you supposed to know what the line is about otherwise?  Leaving it up to the individual module other to include or not include peer is also problematic. 

Can you clarify that this will actually change?  If it just changes the `RHOSTS` `file://` use-case, 👍.   If this will impact any `RHOSTS` use-case such that the peer information will not be printed in statusy messages by default, 👎 .

**Comment by wvu-r7** [comment]
@jhart-r7: I don't think any of us will disagree that it's useful. It's just been one of our most problematic PRs due to the regressions. It was a very aggressive changeset that we have to live with now. We must tread lightly.

**Comment by TheNaterz** [comment]
Sure, so specifically, this change ONLY removes print_prefix, it does not remove peer. If a module uses peer for any specific message, those are unaffected.

A tcp peer is the entire scope that you set in RHOSTS for that module (single, comma-separated, ranges, file) and placing this peer in a print causes an unnecessary readability issue.

My guess is that other uses of peer deal with single hosts, which is fine because it's not an issue at that point.

**Comment by busterb** [comment]
How about this:

```
  def print_prefix
    if rhost && rhost.split(' ').length < 2
      super + peer + ' - '
    else
      super
    end
  end
```

The point of the 'rhost' check was to see if we are in the context of a host or not. Sometimes 'rhost' ends up mapping to 'rhosts' which ends up being too long. This figures it out and does the right thing that you would expect.

**Comment by TheNaterz** [comment]
Seems OK to me!

**Comment by busterb** [comment]
Output of the above change:

```
msf5 auxiliary(scanner/ssh/ssh_version) > run

[+] 192.168.56.1:22       - SSH server version: SSH-2.0-OpenSSH_7.6 ( service.version=7.6 service.vendor=OpenBSD service.family=OpenSSH service.product=OpenSSH service.cpe23=cpe:/a:openbsd:openssh:7.6 service.protocol=ssh fingerprint_db=ssh.banner )
[+] 192.168.56.1:22       - SSH server version: SSH-2.0-OpenSSH_7.6 ( service.version=7.6 service.vendor=OpenBSD service.family=OpenSSH service.product=OpenSSH service.cpe23=cpe:/a:openbsd:openssh:7.6 service.protocol=ssh fingerprint_db=ssh.banner )
^C[*] Caught interrupt from the console...
[*] Auxiliary module execution completed
```

Before any of these changes, MSF would print all of the hosts on the 'completed' line (annoying). Simply removing print_prefix here causes it to have the effect of no host info. This meets in the middle and probably does what was intended all along.

**Comment by wvu-r7** [comment]
Haven't tested it yet, but that looks like intended behavior!

**Comment by 7043mcgeep** [comment]
Is this issue #8812?
